### ISSUE TITLE
Album Cover Not Displaying in Windows Media Player 

### ISSUE THREAD

**Issue by WittyWidget** [justin025/onthespot#46]
###Album Cover Not Displaying in Windows Media Player

**Description:**
The embedded album cover in MP3 files downloaded from Onthespot does not display in Windows Media Player, but it shows correctly in VLC. It also doesn't appear in the Windows File Explorer details pane. This issue is only happening with files downloaded from Onthespot; other MP3 files from different sources display correctly. I have checked the files with MP3Tag, and the album cover is embedded.

**Steps to Reproduce:**
1. Download MP3 files using Onthespot.
2. Open in Windows Media Player.
3. Check for album cover display.
4. Click on the MP3 file in Windows File Explorer to view the details pane and check for album cover display.

**Expected Behavior:**
Album cover should display in both Windows Media Player and Windows File Explorer.

**Environment:**
- OS: Windows 11
- Windows Media Player 
- VLC 

Thank you!

**Comment by 0rkaM** [comment]
I personally cannot reproduce on Win10 and Linux, so maybe is limited to Win11.
Which app version are you running?
Do you have Embed Cover checked in the _Settings Tab_, (I also have _Save Cover_, but that should not help or impair)
I also have installed K-Lite Codec Pack, which sometime fixes a lot of problem and give more control on what your system can code/decode: https://codecguide.com/download_kl.htm.

**Comment by WittyWidget** [comment]
I am using Windows 11 and have tested it on multiple PCs, but I'm experiencing the same issue. I checked the saved cover, but that doesn’t matter since it saves externally. The issue lies with the embedded cover in the metadata. To reproduce the issue, you can download any audio file from another source that has metadata and compare it with both the OntheSpot audio file and the externally downloaded audio file.

**Comment by WittyWidget** [comment]
I downloaded audio files using OntheSpot, but it is not displaying the embedded album cover in the Windows Explorer details pane and in Media Player.

![OntheSpot](https://github.com/user-attachments/assets/be6d9972-7065-4ce0-a032-6915b22ea516)

In contrast, I downloaded files using Spotify Downloader, which correctly displays the embedded album cover in both the Windows Explorer details pane and Media Player.

![Spotify Downloader](https://github.com/user-attachments/assets/26d8b692-ff4e-49af-9e45-aba1f42c88b1)

This issue appears to be specific to OntheSpot, as I have verified that my other local audio files display correctly.

**Comment by 0rkaM** [comment]
If I download from another source the metadata most likely is going to be different, not only depending on format, but also on the same format, so I cannot really compare them to be equal.
If I search the above title, I get 20+ results and 8 of them relate to the Titanic, so to make sure we are checking the same file, could you update with the actual link of the Song you are trying to download.

**Comment by justin025** [comment]
Could you send the link of 'Spotify Downloader', the only desktop application i've seen downloads from youtube.

**Comment by WittyWidget** [comment]
@justin025 Here's the [Spotify Downloader](https://github.com/WilliamSchack/Spotify-Downloader/releases). Yes, it downloads from YouTube using Spotify API, but I am pointing out the issue with the metadata album cover.

**Comment by Coolcraft8** [comment]
I'm having the same type of issue, with the old onthespot I would import the songs into itunes and the album cover would be displayed with the songs using this the album cover doesn't follow over.

**Comment by 0rkaM** [comment]
This is strange, because on my end, I get the app to download the album image and also embed the image correctly.
I am going to repeat the request for the URL link of any of the songs in question... just to be sure I am checking the same thing.

I wonder if it specific to a combination of the "advanced" settings.

Also, to @WittyWidget '_Spotify Downloader_' does not download from Spotify but get the info from Spotify and download from other sources, so using that does not really get you a Spotify Audio file.

**Comment by WittyWidget** [comment]
@0rkaM, I know but I am just giving you an example about the issue comparing both audio files metadata functioning.

 I found that the issue may be related to the handling of album art in ID3v2.4 tags by Windows OS. Specifically, adding an ID3v1 tag alongside the ID3v2.4 tag resolved the problem of album art not displaying in Windows Media player and Explorer’s details pane. The album art now shows correctly when both tags are present. 

Could you fix and release it?

![image](https://github.com/user-attachments/assets/6de7d8ff-7482-45fa-84f0-da4039399291)

**Update**: I just cross-checked, and I have other local audio files that only have ID3v2.4 tags, which work fine on Windows. Maybe the issue is related to the encoder or the embedding of the album cover?

**Comment by 0rkaM** [comment]
TLDR:
The problem is that, even if ID3v2.4 was published on November 2000, not only WMP did not have a correctly functional support for it until 2017, but even today there are still a lot of issues with WMP reading/and writing them correctly without corrupting the whole thing. This is to the point that in every forum I checked, if someone really need or insist to keep WMP as the player of choice, it is suggested to revert to ID3v2.3, or ID3v1.1, but the last at a loss of many features.

That is why I personally use anything else than WMP for just pleasure easy listening: Open Source MPC, FooBar2000, VLC, Kodi, Plex, RockBox mods, Phone, Tablets, etc.

All revisions are backwards compatible, but major versions are not. If a software only supports ID3v2 and below, it will ignore any tags that are version three or higher, and vice-versa; also, if it supports multiple versions only the higher version will be used.
So, if you do not have a specific hardware, or software that does need ID3v1, it is suggested to make away with that and just use ID3v2.


Long Version:

Historically every version of WMP had and still today have problems with each version of ID3 Tags: when ID3 Tags Version 1.0 was created in 1996 and then perfected to v1.1 in 1997, even if it became the de-facto standard everywhere, Microsoft in its infinite wisdom did not have proper support for any of it until way later with WMP 6.1-7.0 and it actually started pushing for the very buggy, v1.2 of 2002-3 variant proposed by BirdCage Software, which nobody wanted to follow: even today the most supported Version 1 still is ID3v1.1.

Remember at that time Microsoft was pushing for its WMA WMV WPL formats parts of ASF and it was trying to dissociate themselves and move people away from MP3 or any other format associated with the media sharing like Limewire and other player that supported the formats like Winamp.

I am not sure if it still true, but let's remember that WMP used to no play any MP3 files that contained any version of compressed [ID3] (https://en.wikipedia.org/wiki/ID3) headers ("tags"), trying to do so resulted in a "The input media file is invalid"!!

Version 2.0 came out 1998 WMP did not followed that specification until way later and still all the way to 2015 WMP had problems with v2.2 and v2.3, they finally fixed in 2017-18 and WMP can now use v2.3 correctly... but not above... it supposed to, but NO: for example it has problem with problems with usage of UTF-16 and UTF-8, rewriting metadata in the wrong field and corrupting some other, and as you can see cannot display the album images. 

My suggestion it to to make sure that you installed all the OS updates and latest WMP, and/or maybe try a different player, to see if it differs... K-Lite codec pack has a great player in it... and try the latest OnTheSpot build.

I personally could make with just ID3v1 in few cases where only listening in screenless  (headless) devices is needed, and where the software/hardware actually rely on external purpose built custom database, but I do need ID3v2 for its features for all other hardware and software devices I use, so it is a must for me, and I am sure the same is for many other people using audio files, and tagging along with v1 have many time created its own set of problems... YMMV.

Cheers.

**Comment by justin025** [comment]
The latest release also now supports asf and other windows media player formats (depending on your build of ffmpeg). You can try experimenting with that.

**Comment by WittyWidget** [comment]
@justin025 It's working now. By the way, there's an issue with the latest release: the bitrate was changed to 320 kbps, while it was previously limited to 160 kbps for free Spotify accounts.

**Comment by justin025** [comment]
![2024-11-10_07-51](https://github.com/user-attachments/assets/45dfd12c-5e30-4206-8137-d60929cd5692)
I added this right here because users wanted to customize download outputs especially considering deezer outputs lossless tracks. Change it to whatever you prefer

**Comment by justin025** [comment]
there is a toggle in 1.0.3 to decide whether you prefer a win10 compatible format.

**Comment by WittyWidget** [comment]
@justin025 Great, I will check ASAP. BTW, I would like to download a playlist with 1500 tracks. Is it safe with the default settings, or what settings would you recommend? I don't mind if it takes more time, but I want to keep my account safe.

**Comment by justin025** [comment]
As of right now no users have reported banned accounts. I've been personally developing and mass downloading on my personal account so I would say its relatively safe. As for recommended settings there is a user in discord who has downloaded over 150,000 songs using the app and knows more about this than me but he generally recommends using a 35 second download delay to avoid rate limits. If you have two accounts you can check rotate active account and the app will alternate between downloading from both.

Let me know if you have any questions.

**Comment by 0rkaM** [comment]
> ...I would like to download a playlist with 1500 tracks. Is it safe with the default settings, or what settings would you recommend? I don't mind if it takes more time, but I want to keep my account safe.

Safe settings can depend a little bit on your prior usage "style" and a bit more on your actual location.

Using the original default setting I was rate-limited pretty quickly, so I used Download Delay = 30 sec and after about a week of 15 hours per day usage, I received noticed to reset my password for 'unusual activity'.

I then used Download Delay = 45, it was fine at home, but I then received the same notice, after I used it on my laptop while airport hopping for work.

After which 45 sec would get me rate-limited at home, but not 'unusual activity' notice.

So, now I only use at home with Download Delay = 93 and Recoverable Download Retry Delay = 71 and I had no problem for 30+ days.

Those settings perhaps are a bit much, but it is still equivalents to just a tiny less than 1000 songs in 24 hours, therefore since it is only for my own use, and the result it is way faster than I can listen to and work on them anyway, plus it is saving me any more PITA, I am fine with that... but you can try anything in between 35 and my 93.
